testing/trino-plugin-reader/src/main/java/io/trino/server/PluginReader.java

```diff
@@ -107,7 +110,9 @@ public Integer call()
                 .collect(toMap(plugin -> plugin.getClass().getName(), identity()));
         Stream<Map.Entry<String, String>> modulesStream = requireNonNull(modulesToPlugins).entrySet().stream();
         if (impactedModules.isPresent()) {
-            List<String> nonPluginModules = impactedModules.get().stream().filter(module -> !modulesToPlugins.containsKey(module)).collect(Collectors.toList());
+            List<String> nonPluginModules = impactedModules.get().stream()
```

Comment:
Probably best to rename this variable since it's no longer just `nonPluginModules`.

Reply:
Also add context to commit message of " Allow plugin reader to ignore dependents of plugins "

IIUC this means we now apply impact analysis in product tests if change is in core/trino-server or core/trino-server-rpm whereas earlier we'd have just run all tests.

Reply:
That brings up concern - what happens when in future some plugin module gets used as dependency in other plugin module?

It might not be as concerning since the only impact would be we'd run more product tests than needed?